src/resource/worker-control.c

```diff
@@ -1027,7 +1033,10 @@ int worker_control_worker_yield(sid_resource_t *res)
 	for (i = 0; i < worker->channel_count; i++) {
 		chan = &worker->channels[i];
 		if (chan->spec->wire.type == WORKER_WIRE_PIPE_TO_PROXY || chan->spec->wire.type == WORKER_WIRE_SOCKET)
-			return _chan_buf_send(chan, WORKER_CHANNEL_CMD_YIELD, NULL);
+			if (!sid_parent_exited)
+				return _chan_buf_send(chan, WORKER_CHANNEL_CMD_YIELD, NULL);
```

Comment:
There can be more channels between the `worker` and `worker_proxy` so we have one more `_chan_buf_send` call (in `worker_channel_send` besides `worker_control_worker_yield`).

To avoid the same check for "parent exited" in all these places, maybe we could just put that check inside `_chan_buf_send`. There we need to check whether we are trying to send through the channel from `worker` side, not `worker_proxy` side (we can use `worker_control_is_worker(chan->owner)`).

Maybe we could also send some unique error code that we can check for in the `_chan_buf_send` callers, like `-ENOLINK`. And when it happens, we can add a log message indicating this condition - so we always know that the `worker` tried to send something through the channel to the parent, but the link is gone...